packages/modules/viewer-react/src/types.ts

```diff
@@ -131,22 +138,43 @@ export interface ViewerInitializerParams extends ViewerIModelAppOptions {
   /** array of iTwin.js Extensions */
   extensions?: ExtensionProvider[];
 }
+export type RequiredViewerProps = XOR<
+  XOR<ConnectedViewerProps, FileViewerProps>,
+  BlankViewerProps
+>;
+
+export type ModelLoaderProps = Partial<
+  ConnectedViewerProps & FileViewerProps & BlankViewerProps
+> &
+  LoaderProps;
+
+export type ViewerProps = RequiredViewerProps & ViewerCommonProps;
+export type ViewerLoaderProps = RequiredViewerProps & LoaderProps;
 
-export interface ConnectedViewerProps {
+export type ConnectedViewerProps = {
   iTwinId: string;
   iModelId: string;
   changeSetId?: string;
-}
-
-export interface FileViewerProps {
+};
+export type FileViewerProps = {
   /** Path to local snapshot or briefcase */
   filePath: string;
-}
+};
 
-export interface BlankViewerProps {
-  blankConnection: BlankConnectionProps;
+// it's fine to say that if either location or extents is defined, then both have to be defined.
+export type BlankViewerProps = {
+  /** @deprecated specify location and extents instead. */
+  blankConnection?: BlankConnectionProps;
   blankConnectionViewState?: BlankConnectionViewState;
-}
+  location?: Cartographic | EcefLocationProps;
+  extents?: Range3dProps;
+  iTwinId?: string;
+};
+
+export type BlankConnectionInitializationProps = {
```

Comment:
should we just move this into iModelViewHelper since no one else uses this?

Reply:
Done